sdltrs: switch to an active fork, build with SDL2

cmitu opened the pull request:
Switched SDLTRS to be built from https://gitlab.com/jengun/sdltrs, which is an active fork of the original SDLTRS emulator.

* use the SDL2 branch, which should be more portable (KMS/Odroid/etc.)
* set a minimum of configuration dir in the default configuration file ($HOME/.sdltrs.t8c)
* although not explicitely specified, the main license seems to be BSD-2-Clause (https://opensource.org/licenses/BSD-2-Clause)

joolswills commented on scriptmodules/emulators/sdltrs.sh:
```diff
@@ -10,46 +10,56 @@
 
 rp_module_id="sdltrs"
-rp_module_desc="Radio Shack TRS-80 Model I/III/4/4P emulator"
+rp_module_desc="Radio Shack TRS-80 Model I/III/4/4P emulator (SDL2 version)"
```
Don't think we need to add SDL2 version - if we are dropping the previous.

joolswills commented on scriptmodules/emulators/sdltrs.sh:
```diff
@@ -10,46 +10,56 @@
+    addEmulator 1 "$md_id-model1" "trs-80" "$md_inst/sdl2trs -fullscreen -nomousepointer -model 1 -romfile $biosdir/level2.rom -showled -diskdir $romdir/trs-80 -disk0 %ROM%"
+    addEmulator 0 "$md_id-model3" "trs-80" "$md_inst/sdl2trs -fullscreen -nomousepointer -model 3 -romfile3 $biosdir/level3.rom -showled -diskdir $romdir/trs-80 -disk0 %ROM%"
+    addEmulator 0 "$md_id-model4" "trs-80" "$md_inst/sdl2trs -fullscreen -nomousepointer -model 4 -romfile3 $biosdir/level4.rom -showled -diskdir $romdir/trs-80 -disk0 %ROM%"
+    addEmulator 0 "$md_id-model4p" "trs-80" "$md_inst/sdl2trs -fullscreen -nomousepointer -model 4p -romfile4p $biosdir/level4p.rom -showled -diskdir $romdir/trs-80 -disk0 %ROM%"
```
if adding more parameters, I think it might be a good time to dedupe this a bit - can use a variable or an array or so.

cmitu replied:
Yes, that would clear things a bit.

joolswills commented on scriptmodules/emulators/sdltrs.sh:
```diff
@@ -10,46 +10,56 @@
 
     [[ "$md_mode" == "remove" ]] && return
 
-    moveConfigFile "$home/sdltrs.t8c" "$md_conf_root/trs-80/sdltrs.t8c"
```
The config location changed or the old one was incorrect ?

If it moved, and is the same format, it should be preserved/migrated.

cmitu replied:
The `sdltrs.t8c` is default on Windows, on Linux is `.sdltrs.t8c (https://gitlab.com/jengun/sdltrs/-/blob/sdl2/src/trs_sdl_interface.c#L926). I assumed it was wrong, but in the RP repo - on which the current version is based - this is actually correct (https://github.com/RetroPie/sdltrs/blob/6c6935a0c730f65185c36b623f2c2f8f6d0ad3cf/src/trs_sdl_interface.c#L900). 
I'll redo this part.

joolswills commented on scriptmodules/emulators/sdltrs.sh:
```diff
@@ -10,46 +10,56 @@
+    iniSet "harddir"    "$romdir/trs-80"
+
+    moveConfigFile "$home/.sdltrs.t8c" "$md_conf_root/trs-80/sdltrs.t8c"
+    chown "$user:$user" "$md_conf_root/trs-80/sdltrs.t8c"
```
This will override any users custom configs for those values - so it should create a config in a temporary location and use "copyDefaultConfig" probably. 

Are all these needed as diskdir is also set on the commandline etc.

cmitu replied:
I don't know why I always think `iniSet` is not overwriting existing values. I'll adjust the configuration part and see if we can get rid of `diskdir` (might be un-necessary).

joolswills replied:
Or it might be better to include it in the config only rather than forcing on commandline - which would be more flexible perhaps for users.

cmitu commented:
@joolswills thanks for the review, I'll adjust the scriptmodule.

cmitu commented:
- I added the changes for migrating the user settings - the new version kept the format and the options.
- emulator commandline - I removed the `diskdir` param, since it's not strictly needed for running the ROM,  `diskdir` is set in the default config to the same folder. I kept the `romfile` parameter, since it allows to specify a romfile for each of the 4 models, while the configuration file only knows about 3 romfiles (i.e. uses `romfile3` for both Model 3 and Model 4 emulation). 
- I removed the symlinks for the `romfile`s, since we pass the full path names to the emulator, they don't need to be in the binary install folder.

I reverted the description change, makes sense.

joolswills commented:
Awesome. Thanks.
